src/Native/Runtime/MathHelpers.cpp

```diff
@@ -82,6 +82,70 @@ EXTERN_C REDHAWK_API UInt64 REDHAWK_CALLCONV RhpDbl2ULng(double val)
     return((UInt64)val);
 }
 
+// CORERT Specific - on Project N the arguments to these helpers are inverted
+#ifdef CORERT
+#include <cmath>
+
+#define CLR_NAN_32 0xFFC00000
+#define CLR_NAN_64 0xFFF8000000000000L
+
+EXTERN_C REDHAWK_API float REDHAWK_CALLCONV RhpFltRem(float dividend, float divisor)
+{
+    //
+    // From the ECMA standard:
+    //
+    // If [divisor] is zero or [dividend] is infinity
+    //   the result is NaN.
+    // If [divisor] is infinity,
+    //   the result is [dividend] (negated for -infinity***).
+    //
+    // ***"negated for -infinity" has been removed from the spec
+    //
+
+    if (divisor==0 || !isfinite(dividend))
+    {
+        UInt32 NaN = CLR_NAN_32;
+        return *(float *)(&NaN);
+    }
+    else if (!isfinite(divisor) && !isnan(divisor))
+    {
+        return dividend;
+    }
+    // else...
+#if 0
+    // COMPILER BUG WITH FMODF() + /Oi, USE FMOD() INSTEAD
+    return fmodf(dividend,divisor);
```

Comment:
I think it's probably fine to use fmodf here. It would be helpful if people hit compiler bugs to say what bug, or at least the version of the compiler. Closest I could find publicly was something from 1999 (http://www.drdobbs.com/bug-of-the-month/184416535)